.ci/pitest.groovy

```diff
@@ -0,0 +1,416 @@
+import groovy.io.FileType
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.Field
+import groovy.xml.XmlParser
+import groovy.xml.XmlSlurper
+import groovy.xml.XmlUtil
+import groovy.xml.slurpersupport.GPathResult
+import groovy.xml.slurpersupport.NodeChildren
+
+@Field final static String SEPARATOR = System.getProperty("file.separator")
+
+final int exitCode
+if (args.length == 2) {
+    exitCode = parseArgumentAndExecute(args[0], args[1])
+}
+else {
+    exitCode = parseArgumentAndExecute(args[0], null)
+}
+System.exit(exitCode)
+
+/**
+ * Parse the command line arguments passed in and execute the branch based on the arguments.
+ *
+ * @param argument command line argument
+ * @return {@code 0} if command executes successfully, {@code 1} otherwise
+ */
+private int parseArgumentAndExecute(String argument, String flag) {
+    final Set<String> profiles = getPitestProfiles()
+    final String usageString = """
+        Usage groovy ./.ci/pitest.groovy [profile] [-g | --generate-suppression]
+        To see the full list of supported profiles run 'groovy ./.ci/pitest.groovy --list'
+        """.stripIndent()
+
+    final int exitCode
+    if (profiles.contains(argument)) {
+        generatePitestReport(argument)
```

Comment:
Why we need to execute  pitest by maven here ?

This script should be just analyze reports. No depending on maven

Reply:
>This is what the previous bash script did.

What is problem with this modular concept ?
If we have problems with script we do not need to return pitest

Reply:
It would break the habit, till now we used to just run `./.ci/pitest.sh [profile]` to do the job for us, now we would add a step to this, it will make it executing difficult locally.

Reply:
> What is problem with this modular concept ?
If we have problems with script we do not need to return pitest

I have no problem with this, just give me a thumbs up and I will separate the script and the maven command.

Reply:
Done, script now just analyzes the report.